src/App.tsx

```diff
@@ -1,26 +1,264 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  createTodo,
+  deleteTodo,
+  getTodos,
+  patchTodo,
+} from './api/todos';
+import { List } from './components/List';
+import { Error as ErrorMessage } from './components/Error';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { ErrorMessages } from './types/ErrorMessages';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState(Filter.All);
+  const [tempTodoTitle, setTempTodoTitle] = useState<string | null>('');
+  const [idsProccesing, setIdsProccesing] = useState<number[]>([]);
+
+  const ref = useRef<HTMLInputElement | null>(null);
```

Comment:
It is not clear for what is this ref, because it has just name ref
So let's think of a descriptive name